sn_networking/src/record_store.rs

```diff
@@ -930,42 +936,17 @@ impl RecordStore for ClientRecordStore {
 // and gives an exponential pricing curve when storage reaches high.
 // and give extra reward (lower the quoting price to gain a better chance) to long lived nodes.
 pub fn calculate_cost_for_records(quoting_metrics: &QuotingMetrics) -> u64 {
-    use std::cmp::{max, min};
-
     let records_stored = quoting_metrics.close_records_stored;
-    let received_payment_count = quoting_metrics.received_payment_count;
     let max_records = quoting_metrics.max_records;
-    let live_time = quoting_metrics.live_time;
-
-    let ori_cost = (10 * records_stored) as u64;
-    let divider = max(1, records_stored / max(1, received_payment_count)) as u64;
-
-    // Gaining one step for every day that staying in the network
-    let reward_steps: u64 = live_time / (24 * 3600);
-    let base_multiplier = 1.1_f32;
-    let rewarder = max(1, base_multiplier.powf(reward_steps as f32) as u64);
-
-    // Fine tuning here helps to get a desired curve:
-    // 1, Close to the max supply (4.3E+18) when stored records reaching full.
-    // 2, Charging around `token`s near the situation of 80% storage reached.
-    //
-    // 1.02.powf(1638) = 1.25E+14 => charge_at_full = 10 * MAX_RECORDS * 1.25E+14 = 5.4E+18
-    // 1.02.powf(820) = 1.1E+7 => charge_at_80_percent = 10 * 0.8 * MAX_RECORDS * 1.1E+7 = 3.6E+11 (360 tokens)
-    let base_multiplier = 1.0225_f32;
-
-    // Given currently the max_records is set at MAX_RECORDS,
-    // hence setting the multiplier trigger at 60% of the max_records
-    let exponential_pricing_trigger = 6 * max_records / 10;
-
-    let multiplier = max(
-        1,
-        base_multiplier.powf(records_stored.saturating_sub(exponential_pricing_trigger) as f32)
-            as u64,
-    );
-
-    let charge = max(10, ori_cost.saturating_mul(multiplier) / divider / rewarder);
-    // Deploy an upper cap safe_guard to the store_cost
-    min(TOTAL_SUPPLY / CLOSE_GROUP_SIZE as u64, charge)
+
+    let calculated_cost = positive_input_0_1_sigmoid(records_stored as f64 / max_records as f64)
+        * MAX_STORE_COST as f64;
+
+    max(MIN_STORE_COST, calculated_cost as u64 + 1)
+}
+
+fn positive_input_0_1_sigmoid(x: f64) -> f64 {
+    1.0 / (1.0 + (-30.0 * (x - 0.5)).exp())
```

Comment:
as per https://chatgpt.com/share/3e22da65-db27-4dbc-8ba0-800a93891331 

looks ot me like -10.0 might be a more desireable curve?

Reply:
with -10, when x = [0...1], the output is [0.0066928509242848554...0.9933233601507481], narrower than the expected [0...1]